### PR TITLE
Preparing for 5.0.0 release

### PR DESCRIPTION
**Pull request by kbevers**
We are getting close to a new release. There's only a few things left to do, most notably resolving #700 which is well on its way. I'd say we should be able to get the first release candidate out the door some time next week.

I have set the release date to 2018-02-15. It is arbitrarily chosen, but I think the final release will be ready around that time. If the date turns out to be different we need to remember to change those.

This PR takes care of much of the bookkeeping related to the release. I have bumped the remaining version numbers, updated the README and NEWS files and updated the HOWTO-RELEASE file since some of the info in it was outdated or missing. Obviously I haven't created new tarballs and other stuff relelated to distribution of the package.

I have not changed the version number in the Sphinx docs since that is already taken care of in #577 which should be merged just before releasing (it updates the website, so merging needs to be coordinated with the release).

I think most of the changes are straight forward but more eyes on this is needed. **One thing in particular that I am unsure about is the SO-version number. I have changed it from "12.0.0" to "12.1.0". This can be debated since I believe there's a few minor changes in `projects.h` that can be interpreted as breaking backwards compatibility (this basically depends on what we define as the API in previous versions). I need your inpout on this.**

Otherwise most of the work has gone into writing up the release notes. I have included everything I could find in PR's and issues that affects users of the package. I might have missed a few minor things, let me know if you stumple on something.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,210 @@
...
+
+ o Adopted terminology from the OGC/ISO-19100 geospatial standards
+   series. Key definitions are:
+   - A coordinate operation is a change of coordinates, based on a
+     one-to-one relationship, from one coordinate reference system
+     to another
+   - A transformation is a coordinate operation in which the two
+     coordinate reference systems are based on different datums, e.g.
+     a change from a global reference frame to a regional frame.
+   - A coordinate operation is an operation in which both coordinate
```
Comment (rouault) [inline NEWS]: a bit confusing: coordinate operation is defined in first and third items

File: NEWS [hunk 2]
```diff
@@ -1,3 +1,210 @@
...
+
+ o Added metadata to init-files that can be read with the
+   proj_init_info() function in the new proj.h API.
+
+ o Added ITRF2000, ITRF2008 and ITRF2014 init-files with ITRF
+   transformation parameters, including plate motion model
+   parameters.
+
+ o Added parameters for GSK2011, PZ90 and danish (similar to existing
+   andrae which has an incorrect semimajor axis) ellipsoids.
```
Comment (rouault) [inline NEWS]: andrae ?

File: NEWS [hunk 3]
```diff
@@ -1,3 +1,210 @@
...
+
+ o Support for analytical partial derivatives has been removed.
+
+ o Improved performance in Winkel Tripel and Aitoff. #406.
+
+
+ BUG FIXES
+ ------------
+
+ o Fixed incorrect convergence calculation in Lamber Conformal
```
Comment (rouault) [inline NEWS]: Lambert

File: NEWS [hunk 4]
```diff
@@ -1,3 +1,210 @@
...
+ o Return correct result at origin in inverse Lee Oblated
+   Stereographic. #164.
+
+ o Return correct result at origin in inverse Miller Oblated
+   Stereographic. #164.
+
+ o Fixed scaling and wrap-around issues in Oblique Cylindrical
+   Equal Area. #166.
+
+ o Corrected a coefficient error in inverse Transform Mercator. #174.
```
Comment (rouault) [inline NEWS]: Transform->Transverse

File: src/Makefile.am [hunk 5]
```diff
@@ -40,7 +40,7 @@ geodtest_LDADD = libproj.la
 
 lib_LTLIBRARIES = libproj.la
 
-libproj_la_LDFLAGS = -no-undefined -version-info 12:0:0
+libproj_la_LDFLAGS = -no-undefined -version-info 12:1:0
```
Comment (rouault) [inline src/Makefile.am]: This should normally be 13:0:1 I think. If I follow correctly the rule in GDAL's HOWTO-RELEASE, in the "adding interfaces" case
```
4) Update LIBGDAL_CURRENT/REVISION/AGE macros in GDALmake.opt.in.
   - For a release with no interface changes just bump REVISION.
   - Adding interfaces, bump CURRENT/AGE, set REVISION to 0.
   - Deleting interfaces / compatibility issues - bump CURRENT, others to zero.
```

File: CMakeLists.txt [hunk 6]
```diff
@@ -49,9 +49,9 @@ colormsg(_HIBLUE_ "Configuring PROJ:")
 #PROJ version information
 #################################################################################
 include(Proj4Version)
-proj_version(MAJOR 4 MINOR 9 PATCH 3)
+proj_version(MAJOR 5 MINOR 0 PATCH 0)
 set(PROJ_API_VERSION "12")
-set(PROJ_BUILD_VERSION "12.0.0")
+set(PROJ_BUILD_VERSION "12.1.0")
```
Comment (rouault) [inline CMakeLists.txt]: I'd say 13.0.1 (see other related comment)

File: NEWS (hunk 1 above)
Comment (kbevers) [reply NEWS]: Oh, I meant to write "conversion", ont "coordinate operation". Good catch.

File: NEWS (hunk 2 above)
Comment (kbevers) [reply NEWS]: Yes, that's a thing:

```
 proj -le | grep andrae
   andrae a=6377104.43     rf=300.0         Andrae 1876 (Den., Iclnd.)
```

Named after Danish geodesist Andræ, but "æ" doesn't play well non-unicode character sets. I'll change it to "andrae ellipsoid" which hopefully is less confusing.

File: src/Makefile.am (hunk 5 above)
Comment (kbevers) [reply src/Makefile.am]: Yes, this is a bit easier to understand. After reading the rules in the PROJ HOWTO-RELEASE i interpret it the same way. 13.0.1 it is.

**Comment by kbevers** [comment]
Updated according to @rouault's comments. ABI version is now set to `13.0.1`.

**Comment by kbevers** [comment]
I few other things I am thinking about adding to the NEWS file:

* Something about naming conventions re: #735. If some sort of consensus is reached at least.
* Some sort of description of the release schedule as described in #686. I think it is only fair to be up front about the expected time schedule for the announced removal of deprecated modules.
* Should we also announce the removal of the COM-modules? I've enquried the mailing list about it's status but so far haven't gotten any feedback. If no one replies I take it as it is unused. I'd remove it with the 5.1.0 release similar to the WinCE module.

**Comment by busstoptaktik** [comment]
A few relevant mailing list discussions:

[Current master...](http://osgeo-org.1560.x6.nabble.com/Current-master-still-at-released-4-9-3-td5336300.html)

[Coming releases...](http://osgeo-org.1560.x6.nabble.com/Coming-releases-of-PROJ-4-td5325216.html)

Also, someone somewhere had a request for some specific numbering scheme for release candidates It might have been @rsbivand or @sebastic but I cannot find the actual comment.

In general, I think we need input on this subject, from people packaging PROJ: We developers are  probably unaware of a number of ways we can mess up things for the people doing packaging, testing and distribution.

Also note [this message](http://lists.maptools.org/pipermail/proj/2017-June/007710.html) from @rsbivand, mentioning the rproj use of a "detect invertibility" function, that will no longer work as of 5.0.0. May be worth mentioning somewhere - or perhaps it should just be filed as an issue on rproj?

**Comment by kbevers** [comment]
> Also note this message from @rsbivand, mentioning the rproj use of a "detect invertibility" function, that will no longer work as of 5.0.0. May be worth mentioning somewhere - or perhaps it should just be filed as an issue on rproj?

Are you talking about this test:

```
#include <projects.h>

int inversetest(PJ *P){
  return (P->inv ? 1: 0);

}
```

If so, it is still possible in 5.0.0. From 6.0.0 `projects.h` will be removed from the public API though.

**Comment by busstoptaktik** [comment]
No, I'm talking about the PJ->inverted being set

Den 25. jan. 2018 4.02 PM skrev "Kristian Evers" <notifications@github.com>:

**Comment by kbevers** [comment]
Ah okay. How likely is that to happen if you are using the old API(s)? I suspect not very

**Comment by busstoptaktik** [comment]
The old APIs can still be initialized through EPSG style definition files, where the actual definition text is opaque to the user, so I still think it worth to warn, and provide an API workaround.

Actually, it seems, we also have the problem in pj_transform.c around line 158.

**Comment by kbevers** [comment]
Sure, we can mention it in the release notes, no problem. We could also make a pj_has_inverse function and expose it in proj_api.h. Should be quite simple to do.

**Comment by rsbivand** [comment]
This is not the only problem in rgdal - I am not the maintainer of the proj4 package which has not been revised since 2012. Once we've checked, I'll post separate issues on breaking changes. @edzer please note sf consequences.

**Comment by edzer** [comment]
Thanks, @rsbivand - I stumbled in a much earlier phase, see #745

**Comment by kbevers** [comment]
> I did note that "make test" fails on testIGNF. I didn't get a chance to figure out why. If something needs to be done before this test can be successfully run, perhaps this could be checked when cmake is configured?

They seem to work on CI so I am not too worried about that. 

> Finally, the presence of -std=c89 with gnu compiles causes the C99_MATH test to fail. I'm OK with this. But why not support c99 if it's available? The C99_MATH usage is supposed only to affect how proj is compiled. Client packages shouldn't need to know about this.

I think it is unrelated to this PR, but please make the changes you feel are necessary. I have no strong opinions about this subject.

**Comment by cffk** [comment]
> > I did note that "make test" fails on testIGNF. I didn't get a chance to figure out why. If something needs to be done before this test can be successfully run, perhaps this could be checked when cmake is configured?

> They seem to work on CI so I am not too worried about that.

Surely, the tests should work for someone trying to build the code?  I'll try to figure out why this one isn't working.

In fact, there's another glaring problem with most of the tests -- all, save geodesic-test, require write access to the source directory.  This is a bad practice.  I'll see about fixing this too.

> > Finally, the presence of -std=c89 with gnu compiles causes the C99_MATH test to fail. I'm OK with this. But why not support c99 if it's available? The C99_MATH usage is supposed only to affect how proj is compiled. Client packages shouldn't need to know about this.

> I think it is unrelated to this PR, but please make the changes you feel are necessary. I have no strong opinions about this subject.

I'm happy to drop this (too close to the release date to be mucking with compiler flags).

**Comment by kbevers** [comment]
> Surely, the tests should work for someone trying to build the code? I'll try to figure out why this one isn't working.

Sorry I should have elaborated on that. What I meant was that usually when the tests fail locally it is because you are missing grids or some other configuration. If that isn't case then indeed it should be fixed.

> In fact, there's another glaring problem with most of the tests -- all, save geodesic-test, require write access to the source directory. This is a bad practice. I'll see about fixing this too.

Good point. I guess it should be possible to write to a temporary location instead. Just for the record, the gie tests doesn't need write access either.

**Pull request merged**